sdk/src/azure/core/az_context.c

```diff
@@ -33,6 +35,8 @@ AZ_NODISCARD int64_t az_context_get_expiration(az_context const* context)
 AZ_NODISCARD az_result
 az_context_get_value(az_context const* context, void const* key, void const** out_value)
 {
+  _az_PRECONDITION_NOT_NULL(context);
+
```

Comment:
```suggestion
if (key == NULL)
{
  *out_value = NULL;
  return AZ_ERROR_ITEM_NOT_FOUND;
}

```

Reply:
Key being a precondition should not all be a runtime check.  The precondition implies the application should never ever do this.  Where as the runtime check says its ok for the value to be something and we will take another path in that case.  We want the former behavior that the preconditions provide.